### ISSUE TITLE
Error of "memoryMB" reported when running pytorch experiment in kubeflow mode

### ISSUE THREAD

**Issue by qtz93** [microsoft/nni#2304]
**Short summary about the issue/question**:
The kubeflow version I use is 0.6. When the version number of pytorch-operator is specified as "v1beta1" or "v1alpha2" in the configuration file, the following errors are reported.

```shell
# NNI WebUI
Job management error: Cannot read property 'memoryMB' of undefined

# NNI log
[4/14/2020, 5:00:04 PM] ERROR [ 'TypeError: Cannot read property \'memoryMB\' of undefined\n    at KubeflowTrainingService.prepareKubeflowConfig (/soft/anaconda3/envs/automl/nni/training_service/kubernetes/kubeflow/kubeflowTrainingService.js:225:94)\n    at KubeflowTrainingService.submitTrialJob (/soft/anaconda3/envs/automl/nni/training_service/kubernetes/kubeflow/kubeflowTrainingService.js:73:46)\n    at process._tickCallback (internal/process/next_tick.js:68:7)' ]
```

However, when using the v1 version of pytorch-operator, the NNI experiment cannot start, as follows:

```shell
(automl) [root@jaserver2 mnist-pytorch]# nnictl create --config config_kubeflow.yml --port 46001
INFO: expand searchSpacePath: search_space.json to /home/.../trials/mnist-pytorch/search_space.json 
INFO: expand codeDir: . to /home/.../trials/mnist-pytorch/. 
INFO: Starting restful server...
INFO: Successfully started Restful server!
INFO: Setting kubeflow config...
ERROR: Failed! Error is: {"error":"Invalid pytorch-operator apiVersion v1"}
```

What is the required Kubeflow version if I run the PyTorch experiment in NNI Kubeflow mode? Thanks~

**Brief what process you are following**: 
Above.

**How to reproduce it**: 

**nni Environment**:
- nni version: 1.3
- nni mode(local|pai|remote): Kubeflow
- OS: CentOS7.5
- python version: 3.6.7
- is conda or virtualenv used?: conda 
- is running in docker?: docker

**need to update document(yes/no)**: yes

**Anything else we need to know**: yes

**Comment by SparkSnail** [comment]
Hi @qtz93. According to `Job management error: Cannot read property 'memoryMB' of undefined`, did you set memoryMB in your config file?   
NNI has support pytorch-v1-operator in v1.5, you could upgrade NNI to latest version and try using it.

**Comment by qtz93** [comment]
@SparkSnail Yes, I set the value of memoryMB to 4096. Thank you for your advice, I'll try version 1.5.

**Comment by qtz93** [comment]
@SparkSnail Hello, I modified the nni version in the container to 1.5, set the pytorch-operator version to v1 in the yml configuration file, and re-executed the pytorch experiment. The following error still occurred: 

```shell
(automl) [root@jaserver2 mnist-pytorch]# nnictl create --config config_kubeflow.yml --port 46001
INFO: expand searchSpacePath: search_space.json to /home/.../trials/mnist-pytorch/search_space.json 
INFO: expand codeDir: . to /home/.../trials/mnist-pytorch/. 
INFO: Starting restful server...
INFO: Successfully started Restful server!
INFO: Setting kubeflow config...
ERROR: Failed! Error is: {"error":"Invalid pytorch-operator apiVersion v1"}
```

Do I need to install version 0.7 of Kubeflow to run the PyTorch experiment? The Kubeflow version I currently use is 0.6.

**Comment by SparkSnail** [comment]
@qtz93, this error message `{"error":"Invalid pytorch-operator apiVersion v1"}` is thrown from NNI, did you upgrade your NNI version to v1.5 in local machine(the machine you run nnictl command)?

**Comment by qtz93** [comment]
@SparkSnail, Well, I just updated nni from 1.3 to 1.5 on the local machine, and found the command "nnictl create --config ..." can be executed successfully, but the initial error appeared again on the NNI interface:

```shell
# NNI WebUI
Job management error: Cannot read property 'memoryMB' of undefined
```

The file config_kubeflow.yml looks like this:

```shell
authorName: default
experimentName: example_dist_pytorch
trialConcurrency: 3
maxExecDuration: 1h
maxTrialNum: 10
#choice: local, remote, pai, kubeflow
trainingServicePlatform: kubeflow
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
nniManagerIp: "172.16.xx.xx"
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner, GPTuner
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  codeDir: .
  worker:
    replicas: 1
    command: python3 mnist.py
    gpuNum: 0
    cpuNum: 1
    memoryMB: 4096
    image: 172.16.xxx.xxx:5000/qiutzh/miniconda_py36_nni_v15:cpu 
kubeflowConfig:
  operator: pytorch-operator
  apiVersion: v1
  storage: nfs
  nfs:
    server: 172.16.xx.xx
    path: /opt/data2/nfs_nni_share
```

--__--. The error seems to have nothing to do with the version of nni? Why?

**Comment by qtz93** [comment]
@SparkSnail, Well, I just updated nni from 1.3 to 1.5 on the local machine, and found the command "nnictl create --config ..." can be executed successfully, but the initial error appeared again on the NNI interface:

```shell
# NNI WebUI
Job management error: Cannot read property 'memoryMB' of undefined
```

The file config_kubeflow.yml looks like this:

```shell
authorName: default
experimentName: example_dist_pytorch
trialConcurrency: 3
maxExecDuration: 1h
maxTrialNum: 10
#choice: local, remote, pai, kubeflow
trainingServicePlatform: kubeflow
searchSpacePath: search_space.json
#choice: true, false
useAnnotation: false
nniManagerIp: "172.16.xx.xx"
tuner:
  #choice: TPE, Random, Anneal, Evolution, BatchTuner, MetisTuner, GPTuner
  builtinTunerName: TPE
  classArgs:
    #choice: maximize, minimize
    optimize_mode: maximize
trial:
  codeDir: .
  worker:
    replicas: 1
    command: python3 mnist.py
    gpuNum: 0
    cpuNum: 1
    memoryMB: 4096
    image: 172.16.xxx.xxx:5000/qiutzh/miniconda_py36_nni_v15:cpu 
kubeflowConfig:
  operator: pytorch-operator
  apiVersion: v1
  storage: nfs
  nfs:
    server: 172.16.xx.xx
    path: /opt/data2/nfs_nni_share
```
I built a private warehouse, and running the TensorFlow example in Kubeflow mode based on the private warehouse is no problem.

--__--. The error seems to have nothing to do with the version of nni? Why?

**Comment by SparkSnail** [comment]
Hi @qtz93, while tf-operator has two role: `ps` and `worker`, pytorch-operator also has two role: `master` and `worker`. In pytorch mode, the `master` is required field, and the `worker` is optional. I noticed that you only set `worker` in your config file, if your job is not distributed job, you can use `master` instead of `worker`. There is a example in NNI:
https://github.com/microsoft/nni/blob/master/examples/trials/mnist-distributed-pytorch/config_kubeflow.yml.
However, seems that NNI does not provide valid error message for users, your issue has given NNI a good feedback to improve.

**Comment by qtz93** [comment]
@SparkSnail Um, thanks for reminding me, I tried the following:

First, I switched back to the 1.3 version of NNI and ran the mnist-distributed-pytorch example, and found that the v1beta1 and v1alpha2 versions of pytorch-operator had the following errors:

```shell
## NNI WebUI
# Job management error: 404 page not found

## NNI log
# [4/16/2020, 9:23:07 AM] INFO [ 'submitTrialJob: form: {"sequenceId":0,"hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"learning_rate\\": 0.0001, \\"momentum\\": 0.4}, \\"parameter_index\\": 0}","index":0}}' ]
# [4/16/2020, 9:23:07 AM] ERROR [ 'Error: 404 page not found\n\n    at _request (/soft/anaconda3/envs/automl/nni/node_modules/kubernetes-client/lib/backends/request.js:189:25)\n    at Request.request [as _callback] (/soft/anaconda3/envs/automl/nni/node_modules/kubernetes-client/lib/backends/request.js:148:14)\n    at Request.self.callback (/soft/anaconda3/envs/automl/nni/node_modules/request/request.js:185:22)\n    at Request.emit (events.js:182:13)\n    at Request.<anonymous> (/soft/anaconda3/envs/automl/nni/node_modules/request/request.js:1161:10)\n    at Request.emit (events.js:182:13)\n    at IncomingMessage.<anonymous> (/soft/anaconda3/envs/automl/nni/node_modules/request/request.js:1083:12)\n    at Object.onceWrapper (events.js:273:13)\n    at IncomingMessage.emit (events.js:187:15)\n    at endReadableNT (_stream_readable.js:1094:12)' ]
```

Then, I switched to the 1.5 version of NNI, and used the v1 version of pytorch-operator, and found that mnist-distributed-pytorch can be successfully started.
NNI logs are as follows:

```shell
[4/16/2020, 9:37:16 AM] INFO [ 'Datastore initialization done' ]
[4/16/2020, 9:37:16 AM] INFO [ 'RestServer start' ]
[4/16/2020, 9:37:16 AM] INFO [ 'Construct Kubeflow training service.' ]
[4/16/2020, 9:37:16 AM] INFO [ 'RestServer base port is 46001' ]
[4/16/2020, 9:37:16 AM] INFO [ 'Rest server listening on: http://0.0.0.0:46001' ]
[4/16/2020, 9:37:18 AM] INFO [ 'NNIManager setClusterMetadata, key: kubeflow_config, value: {"operator":"pytorch-operator","apiVersion":"v1","nfs":{"server":"172.16.xx.xx","path":"/opt/data2/nfs_nni_share"}}' ]
[4/16/2020, 9:37:18 AM] INFO [ 'NNIManager setClusterMetadata, key: nni_manager_ip, value: {"nniManagerIp":"172.16.xx.xx"}' ]
[4/16/2020, 9:37:18 AM] INFO [ 'NNIManager setClusterMetadata, key: trial_config, value: {"codeDir":"/home/.../trials/mnist-distributed-pytorch/.","master":{"replicas":1,"command":"python3 dist_mnist.py","gpuNum":0,"cpuNum":1,"memoryMB":2048,"image":"172.16.xxx.xxx:5000/qiutzh/miniconda_py36_nni_v15:cpu"},"worker":{"replicas":1,"command":"python3 dist_mnist.py","gpuNum":0,"cpuNum":1,"memoryMB":2048,"image":"172.16.xxx.xxx:5000/qiutzh/miniconda_py36_nni_v15:cpu"}}' ]
[4/16/2020, 9:37:18 AM] INFO [ 'Starting experiment: lUdxddQm' ]
[4/16/2020, 9:37:18 AM] INFO [ 'Change NNIManager status from: INITIALIZED to: RUNNING' ]
[4/16/2020, 9:37:18 AM] INFO [ 'Add event listeners' ]
[4/16/2020, 9:37:18 AM] INFO [ 'Run Kubeflow training service.' ]
[4/16/2020, 9:37:18 AM] INFO [ 'RestServer start' ]
[4/16/2020, 9:37:18 AM] INFO [ 'RestServer base port is 46002' ]
[4/16/2020, 9:37:18 AM] INFO [ 'Kubeflow Training service rest server listening on: http://0.0.0.0:46002' ]
[4/16/2020, 9:37:19 AM] INFO [ 'NNIManager received command from dispatcher: ID, ' ]
[4/16/2020, 9:37:19 AM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 0, "parameter_source": "algorithm", "parameters": {"learning_rate": 0.01, "momentum": 0.4}, "parameter_index": 0}' ]
[4/16/2020, 9:37:19 AM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 1, "parameter_source": "algorithm", "parameters": {"learning_rate": 0.001, "momentum": 0.6}, "parameter_index": 0}' ]
[4/16/2020, 9:37:19 AM] INFO [ 'NNIManager received command from dispatcher: TR, {"parameter_id": 2, "parameter_source": "algorithm", "parameters": {"learning_rate": 0.01, "momentum": 0.6}, "parameter_index": 0}' ]
[4/16/2020, 9:37:23 AM] INFO [ 'submitTrialJob: form: {"sequenceId":0,"hyperParameters":{"value":"{\\"parameter_id\\": 0, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"learning_rate\\": 0.01, \\"momentum\\": 0.4}, \\"parameter_index\\": 0}","index":0}}' ]
[4/16/2020, 9:37:23 AM] INFO [ 'submitTrialJob: form: {"sequenceId":1,"hyperParameters":{"value":"{\\"parameter_id\\": 1, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"learning_rate\\": 0.001, \\"momentum\\": 0.6}, \\"parameter_index\\": 0}","index":0}}' ]
[4/16/2020, 9:37:23 AM] INFO [ 'submitTrialJob: form: {"sequenceId":2,"hyperParameters":{"value":"{\\"parameter_id\\": 2, \\"parameter_source\\": \\"algorithm\\", \\"parameters\\": {\\"learning_rate\\": 0.01, \\"momentum\\": 0.6}, \\"parameter_index\\": 0}","index":0}}' ]
[4/16/2020, 9:38:30 AM] INFO [ 'POST: /version/lUdxddQm/vJB0k: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:30 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:30 AM] INFO [ 'POST: /stdout/lUdxddQm/vJB0k: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:30.301831] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:31 AM] INFO [ 'POST: /version/lUdxddQm/f4Yyp: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:31 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:31 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:31.144321] INFO Trial keeper spawns a subprocess (pid 22) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:32 AM] INFO [ 'POST: /version/lUdxddQm/GQX6c: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:32 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:32 AM] INFO [ 'POST: /version/lUdxddQm/vJB0k: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:32 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:32 AM] INFO [ 'POST: /stdout/lUdxddQm/GQX6c: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:32.208243] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:32 AM] INFO [ 'POST: /stdout/lUdxddQm/vJB0k: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:32.209398] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:33 AM] INFO [ 'POST: /version/lUdxddQm/GQX6c: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:33 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:33 AM] INFO [ 'POST: /version/lUdxddQm/f4Yyp: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:33 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 9:38:33 AM] INFO [ 'Trial job vJB0k status changed from WAITING to RUNNING' ]
[4/16/2020, 9:38:34 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:34.427031] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:34 AM] INFO [ 'POST: /stdout/lUdxddQm/GQX6c: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:34.427919] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 9:38:38 AM] INFO [ 'Trial job f4Yyp status changed from WAITING to RUNNING' ]
[4/16/2020, 9:38:38 AM] INFO [ 'Trial job GQX6c status changed from WAITING to RUNNING' ]
[4/16/2020, 9:38:40 AM] INFO [ 'POST: /stdout/lUdxddQm/vJB0k: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:40.340117] INFO subprocess terminated. Exit code is 1. Quit"\n}' ]
[4/16/2020, 9:38:40 AM] INFO [ 'POST: /stdout/lUdxddQm/vJB0k: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:40.348889] INFO NNI trial keeper exit with code 1"\n}' ]
[4/16/2020, 9:38:41 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:41.161167] INFO subprocess terminated. Exit code is 1. Quit"\n}' ]
[4/16/2020, 9:38:41 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 01:38:41.169290] INFO NNI trial keeper exit with code 1"\n}' ]
[4/16/2020, 9:38:41 AM] INFO [ 'POST: /version/lUdxddQm/vJB0k: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 9:38:41 AM] INFO [ 'Version check in trialKeeper success!' ]
...
[4/16/2020, 10:06:38 AM] INFO [ 'POST: /stdout/lUdxddQm/GQX6c: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:38.673059] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 10:06:41 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:41.631900] INFO subprocess terminated. Exit code is 1. Quit"\n}' ]
[4/16/2020, 10:06:41 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:41.639608] INFO NNI trial keeper exit with code 1"\n}' ]
[4/16/2020, 10:06:48 AM] INFO [ 'POST: /stdout/lUdxddQm/GQX6c: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:48.687266] INFO subprocess terminated. Exit code is 1. Quit"\n}' ]
[4/16/2020, 10:06:48 AM] INFO [ 'POST: /stdout/lUdxddQm/GQX6c: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:48.696129] INFO NNI trial keeper exit with code 1"\n}' ]
[4/16/2020, 10:06:57 AM] INFO [ 'POST: /version/lUdxddQm/f4Yyp: body:\n{\n    "tag": "VCSuccess"\n}' ]
[4/16/2020, 10:06:57 AM] INFO [ 'Version check in trialKeeper success!' ]
[4/16/2020, 10:06:57 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:06:57.620575] INFO Trial keeper spawns a subprocess (pid 23) to run command: [\'python3\', \'dist_mnist.py\']"\n}' ]
[4/16/2020, 10:07:07 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:07:07.635872] INFO subprocess terminated. Exit code is 1. Quit"\n}' ]
[4/16/2020, 10:07:07 AM] INFO [ 'POST: /stdout/lUdxddQm/f4Yyp: body:\n{\n    "tag": "trial_keeper",\n    "stdOutputType": "Stdout",\n    "msg": "[2020-04-16 02:07:07.645605] INFO NNI trial keeper exit with code 1"\n}' ]
```

The following errors have occurred in the master and worker in the NFS log:

```shell
# /root/nni/experiments/lUdxddQm/trials-nfs-tmp/nni/lUdxddQm/vJB0k/output/master_output/trialkeeper_stdout:
# /root/nni/experiments/lUdxddQm/trials-nfs-tmp/nni/lUdxddQm/vJB0k/output/worker_output/trialkeeper_stdout:
[2020-04-16 02:06:10.748683] INFO trial_keeper_version is 1.5
[2020-04-16 02:06:10.748783] INFO nni_manager_version is 1.5
[2020-04-16 02:06:10.748803] INFO Version match!
[2020-04-16 02:06:10.763942] INFO Trial keeper spawns a subprocess (pid 23) to run command: ['python3', 'dist_mnist.py']
Traceback (most recent call last):
  File "dist_mnist.py", line 32, in <module>
    from torchvision import datasets, transforms
  File "/opt/conda/lib/python3.6/site-packages/torchvision/__init__.py", line 1, in <module>
    from torchvision import models
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/__init__.py", line 12, in <module>
    from . import detection
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/detection/__init__.py", line 1, in <module>
    from .faster_rcnn import *
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/detection/faster_rcnn.py", line 13, in <module>
    from .rpn import AnchorGenerator, RPNHead, RegionProposalNetwork
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/detection/rpn.py", line 8, in <module>
    from . import _utils as det_utils
  File "/opt/conda/lib/python3.6/site-packages/torchvision/models/detection/_utils.py", line 74, in <module>
    @torch.jit.script
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py", line 364, in script
    graph = _script_graph(fn, _frames_up=_frames_up + 1)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/__init__.py", line 359, in _script_graph
    ast = get_jit_ast(fn)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 132, in get_jit_ast
    return build_def(SourceRangeFactory(source), py_ast.body[0])
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 151, in build_def
    build_stmts(ctx, body))
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 123, in build_stmts
    stmts = [build_stmt(ctx, s) for s in stmts]
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 123, in <listcomp>
    stmts = [build_stmt(ctx, s) for s in stmts]
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 140, in __call__
    return method(ctx, node)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 205, in build_Assign
    rhs = build_expr(ctx, stmt.value)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 140, in __call__
    return method(ctx, node)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 314, in build_Call
    func = build_expr(ctx, expr.func)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 140, in __call__
    return method(ctx, node)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 300, in build_Attribute
    value = build_expr(ctx, expr.value)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 140, in __call__
    return method(ctx, node)
  File "/opt/conda/lib/python3.6/site-packages/torch/jit/frontend.py", line 422, in build_Subscript
    raise NotSupportedError(base.range(), "slicing multiple dimensions at the same time isn't supported yet")
torch.jit.frontend.NotSupportedError: slicing multiple dimensions at the same time isn't supported yet
        proposals (Tensor): boxes to be encoded
    """

    # perform some unpacking to make it JIT-fusion friendly
    wx = weights[0]
    wy = weights[1]
    ww = weights[2]
    wh = weights[3]

    proposals_x1 = proposals[:, 0].unsqueeze(1)
                   ~~~~~~~~~ <--- HERE
    proposals_y1 = proposals[:, 1].unsqueeze(1)
    proposals_x2 = proposals[:, 2].unsqueeze(1)
    proposals_y2 = proposals[:, 3].unsqueeze(1)

    reference_boxes_x1 = reference_boxes[:, 0].unsqueeze(1)
    reference_boxes_y1 = reference_boxes[:, 1].unsqueeze(1)
    reference_boxes_x2 = reference_boxes[:, 2].unsqueeze(1)
    reference_boxes_y2 = reference_boxes[:, 3].unsqueeze(1)

[2020-04-16 02:06:20.778804] INFO subprocess terminated. Exit code is 1. Quit
[2020-04-16 02:06:20.786729] INFO NNI trial keeper exit with code 1

```

There seems to be a problem with the pytorch version. At present, I am using the CPU version of PyTorch (0.4). Which version of PyTorch should I use? Very thanks~

**Comment by qtz93** [comment]
@SparkSnail The problem may be that the versions of torchvision and torch are not compatible. I plan to try to reduce the torchvision version.

**Comment by SparkSnail** [comment]
Hi @qtz93, I noticed that NNI's docker image use `torch==1.2.0` and `torchvision==0.5.0`, you may try it.

**Comment by qtz93** [comment]
@SparkSnail Thank you. I used version 0.2.0 of torchvision, and now the experiment is running normally. By the way, is the current v1 version of pytorch-operator only applicable to NNI1.5?

**Comment by SparkSnail** [comment]
@qtz93 Yes, NNI versions before 1.5 does not support pytorch-operator v1, this feature is recently released in v1.5(latest version).

**Comment by kvartet** [comment]
@qtz93 I’m closing this issue as it has no updates from customer for 3 months, please feel free to reopen if you are still seeing it an active issue.
